stash-origin/image-config.d/10-stash-origin.sh

```diff
@@ -2,3 +2,29 @@
 if [ -f /etc/grid-security/hostcert.pem ]; then
   /usr/local/sbin/fix_certs.sh
 fi
+
+# For the stash-origin image, whether the xrootd and cmsd instances for
+# stash-origin, stash-origin-auth, or both, are selected by environment
+# variables.
+
+# supervisord_... functions are defined in xcache/image-config.d/00-functions.sh
+
+if [[ -n ${XC_AUTH_ORIGIN_EXPORT:-} ]]; then
+  supervisord_enable stash-origin-auth
+  supervisord_enable stash-origin-auth-cmsd
+else
+  supervisord_disable stash-origin-auth
+  supervisord_disable stash-origin-auth-cmsd
+fi
+if [[ -n ${XC_PUBLIC_ORIGIN_EXPORT:-} || -n ${XC_ORIGINEXPORT:-} ]]; then
+  supervisord_enable stash-origin
+  supervisord_enable stash-origin-cmsd
+else
+  supervisord_disable stash-origin
+  supervisord_disable stash-origin-cmsd
+fi
+# backward compat:
+if [[ -z ${XC_AUTH_ORIGIN_EXPORT:-} && -z ${XC_PUBLIC_ORIGIN_EXPORT:-} && -z ${XC_ORIGINEXPORT:-} ]]; then
+  supervisord_enable stash-origin
+  supervisord_enable stash-origin-cmsd
+fi
```

Comment:
Would it make sense to work the backward-compat logic into the supervisord_enable block?  (like below)  Otherwise the logic looks slightly more complicated and does a disable followed by enable.

```suggestion
if [[ -n ${XC_PUBLIC_ORIGIN_EXPORT:-} || -n ${XC_ORIGINEXPORT:-} ]] ||
   # backward compat:
   [[ -z ${XC_AUTH_ORIGIN_EXPORT:-} ]]; then
  supervisord_enable stash-origin
  supervisord_enable stash-origin-cmsd
else
  supervisord_disable stash-origin
  supervisord_disable stash-origin-cmsd
fi
```

Reply:
I think I met you half-way on this: I got rid of the disable-followed-by-enable but I didn't put all the clauses into one if statement because I had to concentrate a bit to read it and verify that it was correct.